torch/nn/init.py

```diff
@@ -0,0 +1,245 @@
+from functools import reduce
+from operator import mul
+import math
+import random
+
+import torch
+from torch.autograd import Variable
+
+
+def uniform(tensor, a=0, b=1):
+    """Fills the input Tensor or Variable with values drawn from a uniform U(a,b)
+
+    Args:
+        tensor: a n-dimension torch.Tensor
+        a: the lower bound of the uniform distribution
+        b: the upper bound of the uniform distribution
+
+    Examples:
+        >>> w = torch.Tensor(3, 5)
+        >>> nninit.uniform(w)
+    """
+    if isinstance(tensor, Variable):
+        uniform(tensor.data, a=a, b=b)
+        return tensor
+    else:
+        return tensor.uniform_(a, b)
+
+
+def normal(tensor, mean=0, std=1):
+    """Fills the input Tensor or Variable with values drawn from a normal distribution with the given mean and std
+
+    Args:
+        tensor: a n-dimension torch.Tensor
+        mean: the mean of the normal distribution
+        std: the standard deviation of the normal distribution
+
+    Examples:
+        >>> w = torch.Tensor(3, 5)
+        >>> nninit.normal(w)
+    """
+    if isinstance(tensor, Variable):
+        normal(tensor.data, mean=mean, std=std)
+        return tensor
+    else:
+        return tensor.normal_(mean, std)
+
+
+def constant(tensor, val):
+    """Fills the input Tensor or Variable with the value `val`
+
+    Args:
+        tensor: a n-dimension torch.Tensor
+        val: the value to fill the tensor with
+
+    Examples:
+        >>> w = torch.Tensor(3, 5)
+        >>> nninit.constant(w)
+    """
+    if isinstance(tensor, Variable):
+        constant(tensor.data, val)
+        return tensor
+    else:
+        return tensor.fill_(val)
+
+
+def _calculate_fan_in_and_fan_out(tensor):
+    if tensor.ndimension() < 2:
+        raise ValueError("fan in and fan out can not be computed for tensor of size ", tensor.size())
+
+    if tensor.ndimension() == 2:  # Linear
+        fan_in = tensor.size(1)
+        fan_out = tensor.size(0)
+    else:
+        num_input_fmaps = tensor.size(1)
+        num_output_fmaps = tensor.size(0)
+        receptive_field_size = reduce(mul, (tensor.numpy().shape[2:]))
```

Comment:
Cool, thanks @alykhantejani ! On a side note, I just realized that this is actually equivalent to `.numel()`, but I think this is good as is.